commit e72194b42ce1921e4b687fbed94e9d199923e77b
Author: Patrick Donnelly <pdonnell@ibm.com>
Date:   Mon Feb 27 15:00:08 2023 -0500

    mgr: rework register_client to take name and replace
    
    Two new parameters:
    
    - "name" to allow setting the name associated with the addrv. Previously
      this was hard-coded to the mgr module name. However, we want mgr
      modules to be able to update the libcephsqlite cluster handle.
    
    - "replace" to replace all addrv for the given "name". libcephsqlite
      only ever has one handle, remove the old ones.
    
    Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

diff --git a/src/mgr/PyModuleRegistry.h b/src/mgr/PyModuleRegistry.h
index e16b2830d91..9af9abb5762 100644
--- a/src/mgr/PyModuleRegistry.h
+++ b/src/mgr/PyModuleRegistry.h
@@ -202,10 +202,14 @@ public:
     return active_modules->get_services();
   }
 
-  void register_client(std::string_view name, entity_addrvec_t addrs)
+  void register_client(std::string_view name, entity_addrvec_t addrs, bool replace)
   {
     std::lock_guard l(lock);
-    clients.emplace(std::string(name), std::move(addrs));
+    auto n = std::string(name);
+    if (replace) {
+      clients.erase(n);
+    }
+    clients.emplace(n, std::move(addrs));
   }
   void unregister_client(std::string_view name, const entity_addrvec_t& addrs)
   {
